packages/legend-studio/src/models/metamodels/pure/model/packageableElements/domain/Association.ts

```diff
@@ -139,13 +139,15 @@ export class Association
       Class,
       `Association property '${property.name}' must be of type 'class'`,
     );
-    assertTrue(
-      deleteEntry(
-        otherPropertyAssociatedClass.propertiesFromAssociations,
-        otherProperty,
-      ),
-      `Can't find property '${otherProperty.name}' from association '${this.path}' in associated class '${otherPropertyAssociatedClass.path}'`,
-    );
+    if (otherPropertyAssociatedClass.propertiesFromAssociations.length) {
```

Comment:
Like I mentioned before, I get this error, but what you're doing here is handling the code smell, you haven't addressed the actual problem, which is the fact that we messed up graph building somewhere.